src/features2d/descriptor_matcher.rs

```diff
@@ -64,75 +49,84 @@ impl DescriptorMatcherType {
+        }
     }
 
     /// Adds descriptors to train a CPU or GPU descriptor collection
-    pub fn add(&mut self, descriptors: &Vec<&Mat>) {
-        let descriptors = descriptors.iter().map(|x| x.inner).collect();
-        let vec_view = CVecView::pack(&descriptors);
+    pub fn add(&mut self, descriptors: impl IntoIterator<Item = &'a Mat>) {
+        let descriptors = descriptors.into_iter().map(|x| x.inner).collect::<Vec<_>>();
         unsafe {
-            cv_matcher_add(self.value, &vec_view);
+            native::cvsys_matcher_add(self.value, descriptors.as_ptr(), descriptors.len());
         }
     }
 
     /// Trains a descriptor matcher
     pub fn train(&mut self) {
-        unsafe { cv_matcher_train(self.value) }
+        unsafe { native::cvsys_matcher_train(self.value) }
     }
 
     /// Returns true if there are no train descriptors
     pub fn is_empty(&self) -> bool {
-        unsafe { cv_matcher_is_empty(self.value) }
+        unsafe { native::cvsys_matcher_is_empty(self.value) }
     }
 
     /// Finds the best match for each descriptor from a query set
     pub fn match_(&self, query_descriptors: &Mat) -> Vec<DMatch> {
-        let mut matches = CVec::<DMatch>::default();
+        let mut matches: native::cvsys_CVec<native::cvsys_DMatch> = unsafe { std::mem::zeroed() };
         unsafe {
-            cv_matcher_match(self.value, query_descriptors.inner, &mut matches);
+            native::cvsys_matcher_match(self.value, query_descriptors.inner, &mut matches);
         }
-        matches.unpack()
+        matches.into()
     }
 
     /// Finds the best match for each descriptor from a query set.
     /// Unlike `match_`, train descriptors collection are passed directly
     pub fn match_two(&self, query_descriptors: &Mat, train_descriptors: &Mat) -> Vec<DMatch> {
-        let mut matches = CVec::<DMatch>::default();
+        let mut matches: native::cvsys_CVec<native::cvsys_DMatch> = unsafe { std::mem::zeroed() };
         unsafe {
-            cv_matcher_match_two(
+            native::cvsys_matcher_match_two(
                 self.value,
                 query_descriptors.inner,
                 train_descriptors.inner,
                 &mut matches,
             );
         }
-        matches.unpack()
+        matches.into()
     }
 
     /// Finds the k best matches for each descriptor from a query set.
     pub fn knn_match(&self, query_descriptors: &Mat, k: usize) -> Vec<Vec<DMatch>> {
-        let mut matches = CVec::<CVec<DMatch>>::default();
+        let mut matches: native::cvsys_CVec<native::cvsys_CVec<native::cvsys_DMatch>> = unsafe { std::mem::zeroed() };
         unsafe {
-            cv_matcher_knn_match(self.value, query_descriptors.inner, k as c_int, &mut matches);
+            native::cvsys_matcher_knn_match(self.value, query_descriptors.inner, k as c_int, &mut matches);
         }
-        matches.unpack()
+        let match_conv = |matches: native::cvsys_CVec<native::cvsys_CVec<native::cvsys_DMatch>>| {
```

Comment:
I think it's better to import some `native` types to make it more readable.

Reply:
Alright, I will see what I can do to improve the readability.